### ISSUE TITLE
GNSS_SDR: add -DENABLE_OSMOSDR=ON to build flags

### ISSUE THREAD

**Issue by Split7fire** [NixOS/nixpkgs#159996]
While working with gnss-sdr found out that `gnss-sdr` package in nixos built without osmosdr support. Looking through [nixos packages](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/applications/radio/gnss-sdr/default.nix#L85) confirm this.
Can we just add add `-DENABLE_OSMOSDR=ON` to cmakeFlags?

**Comment by Mindavi** [comment]
I think that's possible, now building with that flag. How can I best test if it actually works?

**Comment by Split7fire** [comment]
> Note 2: It does at least build with that flag enabled :)

That's great! I think best option to test it is to start gnss-sdr with OsmoSDR clients. So I can test it and reply whether it works or nor.

**Comment by Split7fire** [comment]
@Mindavi checking [nixos packages](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=gnss) and not find OSMOSDR flag. That is the status?

**Comment by Mindavi** [comment]
Sorry for the slow response. I'm not sure how to test it, that's why I didn't enable it. To me it doesn't make sense to add it without testing, but it could be done. It does feel quite safe, even if it doesn't work, it presumably won't break other functionality (but don't quote me on that).

**Comment by Split7fire** [comment]
> I'm not sure how to test it
I know how to test it ) I have HackRF and want to use it with osmosdr. But I just don't understand how to override package with this option. That's why I asked for "test".

**Comment by Mindavi** [comment]
@bjornfor could you have a look at this? I'm not really sure where to begin.

**Comment by bjornfor** [comment]
I don't have that HW but if someone writes a PR to add support I can help with merging.

**Comment by Mindavi** [comment]
Yeah, I tried building with that flag again but now it errored out, so I think it's more complicated than I previously thought it was.

**Comment by chuangzhu** [comment]
Solved by #418364
